cmd/clusterctl/client/config.go

```diff
@@ -54,6 +55,45 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+func (c *clusterctlClient) GenerateProvider(providerName string, providerType clusterctlv1.ProviderType, options ComponentsOptions) (Components, error) {
+
+	providerName, providerVersion, err := parseProviderName(providerName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Gets the provider configuration (that includes the location of the provider repository)
+	providerConfig, err := c.configClient.Providers().Get(providerName, providerType)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get a client for the provider repository and read the provider components;
+	// during the process, provider components will be processed performing variable substitution, customization of target
+	// namespace etc.
+	// Currently we are not supporting custom yaml processors for the provider
+	// components. So we revert to using the default SimpleYamlProcessor.
+	repositoryClientFactory, err := c.repositoryClientFactory(RepositoryClientFactoryInput{Provider: providerConfig})
+	if err != nil {
+		return nil, err
+	}
+
+	latestMetadata, err := repositoryClientFactory.Metadata(providerVersion).Get()
```

Comment:
This command is also supposed to work when a user does not provider a version in the input (Example: `clusterctl generate provider -i aws`). In such cases the `providerVersion` here will be empty and this command will fail. If the user does not provider a version we have to fall back to the default version.

However, I see that the repository client here does not have a `DefaultVersion` function. I would suggest adding a `DefaultVersion` function to the repository.Client interface and to the implementation (`type repositoryClient struct {...}`).
You can have something like this:
```go
func (c *repositoryClient) DefaultVersion() string {
	return c.repository.DefaultVersion()
}
```
@fabriziopandini Thoughts about adding the `DefaultVersion` function to `repository.Client` interface? This interface already has a `GetVersions` function. I dont think it is unreasonable to add a `DefaultVersion` function.


Then, once you have access to the DefaultVersion function, after l.63 you can have to check to see if `providerVersion` is empty and set it to default version. This should fix the broken use case.